src/openzeppelin/tests/test_account.cairo

```diff
@@ -1,38 +1,102 @@
 use array::ArrayTrait;
+use core::result::ResultTrait;
+use debug::PrintTrait;
+use option::OptionTrait;
+use starknet::class_hash::Felt252TryIntoClassHash;
 use starknet::ContractAddress;
 use starknet::contract_address_const;
+use starknet::syscalls::deploy_syscall;
 use starknet::testing::set_caller_address;
 use starknet::testing::set_contract_address;
+use starknet::testing::set_transaction_hash;
+use starknet::testing;
+use starknet::testing::set_signature;
+use traits::TryInto;
+
 use openzeppelin::account::Account;
+use openzeppelin::account::IAccountDispatcher;
+use openzeppelin::account::IAccountDispatcherTrait;
 use openzeppelin::account::ERC165_ACCOUNT_ID;
 use openzeppelin::account::ERC1271_VALIDATED;
+use openzeppelin::account::TRANSACTION_VERSION;
 use openzeppelin::account::Call;
 use openzeppelin::introspection::erc165::IERC165_ID;
 
-fn NEW_KEY() -> felt252 { 0x444444 }
-fn PUBLIC_KEY() -> felt252 { 0x333333 }
-fn OTHER() -> ContractAddress { contract_address_const::<0x222222>() }
-fn ACCOUNT_ADDRESS() -> ContractAddress { contract_address_const::<0x111111>() }
+#[derive(Drop)]
+struct ValidSignature {
+    private_key: felt252,
+    public_key: felt252,
+    message: felt252,
+    r: felt252,
+    s: felt252
+}
 
-fn setup() -> Array::<Call> {
-    let mut CALLS = ArrayTrait::new();
+fn NEW_KEY() -> felt252 {
+    0x444444
+}
+fn PUBLIC_KEY() -> felt252 {
+    0x333333
+}
+fn OTHER() -> ContractAddress {
+    contract_address_const::<0x222222>()
+}
+fn ACCOUNT_ADDRESS() -> ContractAddress {
+    contract_address_const::<0x111111>()
+}
+fn VALID_SIGNATURE() -> ValidSignature {
+    ValidSignature {
+        private_key: 1234,
+        public_key: 883045738439352841478194533192765345509759306772397516907181243450667673002,
+        message: 2717105892474786771566982177444710571376803476229898722748888396642649184538,
+        r: 3068558690657879390136740086327753007413919701043650133111397282816679110801,
+        s: 3355728545224320878895493649495491771252432631648740019139167265522817576501
+    }
+}
+fn CALLS() -> Array::<Call> {
+    let mut calls = ArrayTrait::new();
+
+    calls.append(
+        Call {
+            to: contract_address_const::<123456>(), selector: 0x123, calldata: ArrayTrait::new()
+        }
+    );
 
-    CALLS.append(Call{
-        to: contract_address_const::<123456>(),
-        selector: 0x123,
-        calldata: ArrayTrait::new()
-    });
+    calls
+}
 
+fn setup() -> Array::<Call> {
     set_contract_address(ACCOUNT_ADDRESS());
     Account::constructor(PUBLIC_KEY());
 
-    return CALLS;
+    CALLS()
+}
+
+fn setup_dispatcher() -> IAccountDispatcher {
```

Comment:
why do we need a separate setup function?

Reply:
We don't. I avoided refactoring the rest of the tests while working on getting the first working version, but I will refactor them accordingly.